code/modules/power/apc/apc.dm

```diff
@@ -242,127 +242,152 @@
-
-			user.visible_message(\
-				"[user.name] has inserted the power cell to [name]!",\
-				"<span class='notice'>You insert the power cell.</span>")
-			chargecount = 0
-			update_icon()
 
+		if(stat & MAINT)
+			to_chat(user, "<span class='warning'>[src] has no electronics inside!</span>")
+			return ITEM_INTERACT_COMPLETE
+
+		if(!user.drop_item())
+			return ITEM_INTERACT_COMPLETE
+
+		used.forceMove(src)
+		cell = used
+		user.visible_message(
+			"<span class='notice'>[user] inserts [used] into [src].</span>",
+			"<span class='notice'>You insert [used] into [src].</span>"
+			)
+		for(var/mob/living/simple_animal/demon/pulse_demon/demon in cell)
+			demon.forceMove(src)
+			demon.current_power = src
+			if(!being_hijacked) // First come, first serve!
+				demon.try_hijack_apc(src)
+		if(being_hijacked)
+			cell.rigged = FALSE // Do not explode the demon.
+		chargecount = 0
+		update_icon()
 		return ITEM_INTERACT_COMPLETE
-	else if(used.GetID())			// trying to unlock the interface with an ID card
+
+	// Swiping ID card.
+	if(used.GetID())
 		togglelock(user)
 		return ITEM_INTERACT_COMPLETE
 
-	else if(istype(used, /obj/item/stack/cable_coil) && opened)
+	// Adding cables.
+	if(istype(used, /obj/item/stack/cable_coil) && opened)
 		var/turf/host_turf = get_turf(src)
 		if(!host_turf)
 			throw EXCEPTION("attackby on APC when it's not on a turf")
 			return ITEM_INTERACT_COMPLETE
+
 		if(host_turf.intact)
-			to_chat(user, "<span class='warning'>You must remove the floor plating in front of the APC first!</span>")
+			to_chat(user, "<span class='warning'>You must expose the floor plating in front of [src] first!</span>")
 			return ITEM_INTERACT_COMPLETE
-		else if(terminal) // it already have terminal
-			to_chat(user, "<span class='warning'>This APC is already wired!</span>")
+
+		if(terminal)
+			to_chat(user, "<span class='warning'>[src] is already wired!</span>")
 			return ITEM_INTERACT_COMPLETE
-		else if(!has_electronics())
-			to_chat(user, "<span class='warning'>There is nothing to wire!</span>")
+
+		if(!has_electronics())
+			to_chat(user, "<span class='warning'>[src] has no electronics inside to wire!</span>")
 			return ITEM_INTERACT_COMPLETE
 
 		var/obj/item/stack/cable_coil/C = used
 		if(C.get_amount() < 10)
-			to_chat(user, "<span class='warning'>You need ten lengths of cable for APC!</span>")
+			to_chat(user, "<span class='warning'>You need ten lengths of cable to wire [src]!</span>")
 			return ITEM_INTERACT_COMPLETE
-		user.visible_message("[user.name] adds cables to the APC frame.", \
-							"<span class='notice'>You start adding cables to the APC frame...</span>")
+
+		user.visible_message(
+			"<span class='notice'>[user] starts adding cables to [src]...</span>",
+			"<span class='notice'>You start adding cables to [src]...</span>"
+			)
 		playsound(loc, 'sound/items/deconstruct.ogg', 50, TRUE)
-		if(do_after(user, 20, target = src))
+		if(do_after(user, 2 SECONDS, target = src))
```

Comment:
You'll want to move these durations into variables so that you can set them to zero during tests.